### PR TITLE
Bluetooth: audio: Reject QoS if CIS is already used specific direction

### PR DESCRIPTION
**Pull request by MariuszSkamra**
As per ASCS_v1.0
If a client requests a Config QoS operation for an ASE that would
result in more than one (Sink/Source) ASE having identical CIG_ID
and CIS_ID parameter values for that client, the server shall not
accept the Config QoS operation for that ASE.
The server shall send a notification of the ASE Control Point
characteristic to the client, the server shall set the Response_Code
value for that ASE to 0x09 (Invalid Parameter Value), and the server
shall set the Reason value for that ASE to 0x0A
(Invalid_ASE_CIS_Mapping).

### CODE REVIEW

**Comment by MariuszSkamra** [comment]
@szymon-czapracki FYI

File: subsys/bluetooth/audio/ascs.c [hunk 1]
```diff
@@ -770,53 +770,30 @@ static uint8_t ascs_attr_cb(const struct bt_gatt_attr *attr, uint16_t handle,
...
-			       audio_iso, paired_stream);
+		if (audio_iso->sink_stream == NULL && audio_iso->source_stream == NULL) {
+			free_audio_iso = audio_iso;
+			continue;
+		} else if (audio_iso->sink_stream && audio_iso->sink_stream->ep) {
+			ep = audio_iso->sink_stream->ep;
+		} else if (audio_iso->source_stream && audio_iso->source_stream->ep) {
+			ep = audio_iso->source_stream->ep;
+		} else {
+			/* XXX: Stream not associated with endpoint. Should we assert??? */
```
Comment (Thalley) [inline subsys/bluetooth/audio/ascs.c:792]: Probably not a bad idea

**Review by Thalley — APPROVED** [review]

File: subsys/bluetooth/audio/ascs.c (hunk 1 above)
Comment (sjanc) [reply subsys/bluetooth/audio/ascs.c:792]: can this be somehow triggered by bogus peer device?  if so I'd make it LOG_ERROR/WARN()

**Review by sjanc — APPROVED** [review]

File: subsys/bluetooth/audio/ascs.c (hunk 1 above)
Comment (Thalley) [reply subsys/bluetooth/audio/ascs.c:792]: No, this would be an internal error.

**Pull request merged**